client/my-sites/email/add-mailboxes/index.tsx

```diff
@@ -192,6 +196,10 @@ const MailboxNotices = ( {
 			source,
 		} );
 
+		if ( ! selectedSite ) {
+			throw new Error( 'Cannot finish unused mailbox setup without selected site' );
```

Comment:
Since `emailManagementTitanSetUpMailbox()` requires a site slug, this will prevent redirecting to an invalid URL by throwing an explicit error instead.